### PR TITLE
[Ang Yuze] iP

### PR DESCRIPTION
**Pull request by YZmunchmunch**
# **MUNCH**

> "Eat your sorrows away" - nobody ( [source](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/cb647821-dc91-4178-847f-42c98fbcd407/deogqv8-c614049b-bcf6-440a-9765-9f0b8299e350.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcL2NiNjQ3ODIxLWRjOTEtNDE3OC04NDdmLTQyYzk4ZmJjZDQwN1wvZGVvZ3F2OC1jNjE0MDQ5Yi1iY2Y2LTQ0MGEtOTc2NS05ZjBiODI5OWUzNTAucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.asrcbJKBG2WDOG76pc9cFPMQIZZvMiVkV0Qz47u2_14) )

#### MUNCH frees your mind from the trouble of remembering things you need to do, so that you can focus on eating. It's
- Text-based
- Easy to learn
- **Beautiful** user interface

It's amazing :star_struck: :star_struck:

**All you have to do is,**
1. Download it
2. Click it
3. Add your tasks
4. Have faith in it :pray:
5. Have ur meal :hamburger:

#### MUNCH's features
- [x] Simple todo tasks
- [x] Managing deadlines 
- [ ] Reminders (coming soon)

#### If you are a Java programmer, you can use it to practice Java too! Here's the main method
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
#### So what are you waiting for? Join MUNCH and forget your worries :stuck_out_tongue_winking_eye:

### CODE REVIEW

File: src/main/java/AddTasks/Deadlines.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package AddTasks;
```
Comment (Jarrett0203) [inline src/main/java/AddTasks/Deadlines.java]: Perhaps you could change the package name to all lowercase? Similarly for the `Exceptions` class as well.

File: src/main/java/Munch.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        System.out.println("Hello! I am Munch! :)");
+        System.out.println("How may I help you?");
+        System.out.println("__________________________");
+    }
+
+    public static void wrongDateFormatMessage() {
+        System.out.println("Wrong format for date! [Format: dd/MM/yyyy]");
+    }
+
+    public static LocalDate convert(String date) {
```
Comment (Jarrett0203) [inline src/main/java/Munch.java]: Maybe you can change the method name to `convertDate` to be more specific?

File: src/main/java/AddTasks/Deadlines.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class Deadlines extends Task {
```
Comment (Jarrett0203) [inline src/main/java/AddTasks/Deadlines.java]: Maybe you can change the class name to a singular form? Similarly for the `Events` class as well.

File: src/main/java/FileSave.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        try {
+            if (!storageFile.exists()) {
+                storageFile.createNewFile();
+            } else {
+                storageFile.delete();
+                storageFile.createNewFile();
+            }
+            FileWriter munch = new FileWriter(path);
+            BufferedWriter munchWrite = new BufferedWriter(munch);
+            for(int i = 0; i < tasks.size(); i++) {
```
Comment (Jarrett0203) [inline src/main/java/FileSave.java]: Maybe you can add a space between `for` and `(` ?

File: src/main/java/munch/TaskList.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import munch.Parser;
+
+import java.lang.reflect.Array;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class TaskList {
+
+    public static void run(ArrayList<Task> tasks, String word, String[] words) throws IncompleteInputException, InvalidInputException {
```
Comment (Infrix) [inline src/main/java/munch/TaskList.java]: Perhaps this method is a little too complicated to be in TaskList? Here, the TaskList is performing the job of parsing input, interpreting the command and running the command

An abstract command class could be created for different commands to inherit from, for them to handle their own execution

File: src/main/java/munch/TaskList.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    public static void addTodoTask(ArrayList<Task> tasks, String word) throws IncompleteInputException {
+        String separator = "todo";
+        int sepPos = word.indexOf(separator);
+        String str = word.substring(sepPos + separator.length() + 1);
+        if (str.length() != 0) {
+            Todo todos = new Todo(str);
+            tasks.add(todos);
+            Ui.addTaskMessage();
+            System.out.println(todos);
+            System.out.println("Now you have " + tasks.size() + " task(s) in the list ~~");
```
Comment (Infrix) [inline src/main/java/munch/TaskList.java]: Should the job of printing be handled by the UI instead?

File: src/main/java/munch/TaskList.java (hunk 5 above)
Comment (glyfy) [inline src/main/java/munch/TaskList.java]: Perhaps create an abstract command class to reduce code?

File: src/main/java/munch/TaskList.java (hunk 6 above)
Comment (glyfy) [inline src/main/java/munch/TaskList.java]: should printing be handled by UI class?

**Pull request closed without merging**